docs/docs/data-structures/tree/node-types.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: Node Types
+sidebar_position: 2
+---
+
+The following leaf node types are available:
```

Comment:
This is a very abrupt start. Readers will need some context, which I see was not really provided in the original.
How about:
```suggestion
A SharedTree's data is organized into a tree of nodes.
A leaf node represents an atomic value, while an interior node represent an object or collection.

## Leaf Nodes

The following leaf node types are available:
```